app/users/services/ArchimatchUserService.py

```diff
@@ -153,71 +113,52 @@ def archimatch_user_reset_password(cls, request):
         """
         try:
             data = request.data
-            request_keys = set(data.keys())
-            expected_keys = {
-                "id",
-                "old_password",
-                "new_password",
-                "confirm_new_password",
-            }
-
-            # Validate input keys
-            cls.handle_user_data(request_keys, expected_keys)
+            serializer = ArchimatchUserResetPWSerializer(data=data)
+            serializer.is_valid(raise_exception=True)
+            validated_data = serializer.validated_data
 
             # Check if user exists
-            if not ArchimatchUser.objects.filter(id=data.get("id")).exists():
-                response_data = {
-                    "message": {"message": "User does not exist"},
-                    "status_code": status.HTTP_404_NOT_FOUND,
-                }
-                return Response(
-                    response_data["message"],
-                    status=response_data["status_code"],
-                )
+            user = request.user
+            old_password = validated_data.get("old_password")
+            new_password = validated_data.get("new_password")
+            confirm_new_password = validated_data.get("confirm_new_password")
 
-            user = ArchimatchUser.objects.get(id=data.get("id"))
-            old_password = data.get("old_password")
-            new_password = data.get("new_password")
-            confirm_new_password = data.get("confirm_new_password")
-
-            # Check if old password matches
             if not user.check_password(old_password):
-                response_data = {
-                    "message": {"message": "Incorrect old password"},
-                    "status_code": status.HTTP_400_BAD_REQUEST,
-                }
-                return Response(
-                    response_data["message"],
-                    status=response_data["status_code"],
-                )
-
+                raise ValidationError("Incorrect old password")
             # Set new password if conditions are met
             if new_password == confirm_new_password:
```

Comment:
if new_password != confirm_new_password:
raise ValidationError("New passwords do not match")

Reply:
same thing here : 

```
if new_password == confirm_new_password:
                user.set_password(new_password)
                user.save()
                return Response(
                    {"message": "Password successfully updated"},
                    status=status.HTTP_200_OK,
                )
            else:
                raise ValidationError(detail="New passwords do not match")
```